geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/results/LuceneGetPageFunction.java

```diff
@@ -65,15 +70,19 @@ public void execute(FunctionContext context) {
     }
   }
 
-  protected PageEntry getEntry(final Region region, final Object key) {
+  protected PageEntry getEntry(final Region region, final Object key,
+      SecurityService securityService, Object principal) {
     final EntrySnapshot entry = (EntrySnapshot) region.getEntry(key);
     if (entry == null) {
       return null;
     }
 
-    final Object value = entry.getRegionEntry().getValue(null);
+    Object value = entry.getRegionEntry().getValue(null);
     if (value == null || Token.isInvalidOrRemoved(value)) {
       return null;
+    } else if (securityService.needPostProcess()) {
+      value = securityService.postProcess(principal, region.getFullPath(), key, entry.getValue(),
```

Comment:
I'm a little concerned that `value` here might be a CachedDeserializable in some cases. I think this whole function and PageEntry class are designed to try to pass the value back to the user without deserializing it if it can. 

However I can't think of a case that you didn't test already, so I guess it's doing the right thing.